### PR TITLE
[typescript-migration] test/year_picker_test.test.js and some tests

### PR DESCRIPTION
**Pull request by yuki0410-dev**
## Description
**Linked issue**: #4700

**Changes** `year_picker_test.test.js` and some tests ware migrated to ts

## Contribution checklist
* [x]  I have followed the [contributing guidelines](https://github.com/Hacker0x01/react-datepicker/blob/main/CONTRIBUTING.md).
* [x]  I have added sufficient test coverage for my changes.
* [x]  I have formatted my code with Prettier and checked for linting issues with ESLint for code readability.

### CODE REVIEW

File: test/year_dropdown_options_test.test.tsx [hunk 1]
```diff
@@ -299,24 +302,34 @@ describe("YearDropdownOptions with scrollable dropwdown", () => {
...
     textContents = Array.from(
       container.querySelectorAll(".react-datepicker__year-option"),
     ).filter((node) => node.textContent);
 
-    const x = textContents.find((year) => year === utils.getYear(minDate));
-    expect(x).toBeUndefined();
+    const x = textContents.find(
+      (year) => parseInt(year.textContent ?? "") === utils.getYear(minDate),
+    );
+    expect(x).not.toBeUndefined();
```
Comment (yuki0410-dev) [inline test/year_dropdown_options_test.test.tsx:328]: It seemed that `not` was needed to see the subsequent checklist items.

File: test/year_dropdown_options_test.test.tsx [hunk 2]
```diff
@@ -368,20 +388,26 @@ describe("YearDropdownOptions with scrollable dropwdown", () => {
...
       container.querySelectorAll(".react-datepicker__year-option"),
     ).filter((node) => node.textContent);
 
     expect(
-      textContents.find((year) => year === utils.getYear(minDate)),
-    ).toBeUndefined();
+      textContents.find(
+        (year) => parseInt(year.textContent ?? "") === utils.getYear(minDate),
+      ),
+    ).not.toBeUndefined();
```
Comment (yuki0410-dev) [inline test/year_dropdown_options_test.test.tsx:410]: It seemed that `not` was needed to see the subsequent checklist items.

File: test/year_dropdown_options_test.test.tsx [hunk 3]
```diff
@@ -424,20 +450,26 @@ describe("YearDropdownOptions with scrollable dropwdown", () => {
...
       container.querySelectorAll(".react-datepicker__year-option"),
     ).filter((node) => node.textContent);
 
     expect(
-      textContents.find((year) => year === utils.getYear(maxDate)),
-    ).toBeUndefined();
+      textContents.find(
+        (year) => parseInt(year.textContent ?? "") === utils.getYear(maxDate),
+      ),
+    ).not.toBeUndefined();
```
Comment (yuki0410-dev) [inline test/year_dropdown_options_test.test.tsx:472]: It seemed that `not` was needed to see the subsequent checklist items.

File: test/year_dropdown_options_test.test.tsx [hunk 4]
```diff
@@ -23,9 +25,10 @@ describe("YearDropdownOptions", () => {
   });
 
   it("shows the available years in the initial view", () => {
-    const yearDropdownNode = yearDropdown.querySelector("div");
+    const yearDropdownNode = yearDropdown?.querySelector("div");
     const textContents = Array.from(
-      yearDropdownNode.querySelectorAll(".react-datepicker__year-option"),
+      yearDropdownNode?.querySelectorAll(".react-datepicker__year-option") ??
```
[the remark this answers is not in the document]
Comment (yuki0410-dev) [reply test/year_dropdown_options_test.test.tsx:30]: Thanks for the advice.
I found your advice very idealistic.
However, the purpose of this PR is not to refactor the test, but to TS it, so I am going to leave it as it is this time.
(I hope to refactor it someday)

**Comment by martijnrusschen** [comment]
Nice work!

**Pull request merged**